### PR TITLE
Add rocmcc into the spack compilers package

### PR DESCRIPTION
**Pull request by srekolam**
i ran the unit test 
spack unit-test lib/spack/spack/test/compilers/detection.py
spack unit-test lib/spack/spack/test/compilers/basics.py

tried spack compiler add [path-to-llvm-amdgpu@4.3.0] etc

### CODE REVIEW

**Comment by srekolam** [comment]
@adamjstewart  @haampie  can you please review. I see failures while running  lib/spack/spack/test/architecture.py in the log but i was not sure if i need to do any thing..

**Comment by srekolam** [comment]
@haampie  or @adamjstewart  can i get this reviewed ?

**Comment by srekolam** [comment]
can i get the review comments for this PR

File: lib/spack/spack/compilers/rocmcc.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        if match:
+            ver = match.group(match.lastindex)
+        return ver
+
+    @classmethod
+    def fc_version(cls, fortran_compiler):
+        if sys.platform == 'darwin':
+            return cls.default_version('clang')
+
+        return cls.default_version(fortran_compiler)
```
Comment (adamjstewart) [inline lib/spack/spack/compilers/rocmcc.py:116]: Does this actually apply to this compiler or is this just copy-n-pasted from the clang package?

**Review by adamjstewart — COMMENTED** [review]
I don't know anything about this compiler so I might not be the best person to review. From Spack's perspective everything LGTM.

File: lib/spack/spack/compilers/rocmcc.py (hunk 1 above)
Comment (srekolam) [reply lib/spack/spack/compilers/rocmcc.py:116]: thanks .. this was intended to show up for the amdclang inside the binary package for llvm-amdgpu. i used the existing implementation from aocc.py ..i foresee there will be some new changes in the future.  
The llvm-amdgpu will be renamed as rocm-llvm soon. 
Also the search string for clang will need amd string. Currently based on rocm-4.3.1 it is correct but new release will need r'amd clang version ([^ )\n]+?)-[~.\w\d-]*|' . For fortran too, there will be change.

**Comment by srekolam** [comment]
@becker33  i had this PR some time ago..if you would want to take a look at this and review it let me know.

**Pull request closed without merging**